### PR TITLE
Support log file rolling in jvm presence packaging tests

### LINKED ISSUE #41436: [CI] Tests that are running at midnight and assert on log contents can fail because the log rolls over
There is a failure in https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.0+packaging-tests/306/console that took me a while to understand.

The error is this:

```
1) test42BundledJdkRemoved(org.elasticsearch.packaging.test.OssDebBasicTests)
java.lang.AssertionError: 
Expected: a string containing "/usr/lib/jvm/java-8-openjdk-amd64/jre"
     but: was "[2019-04-23T00:00:02,240][INFO ][o.e.d.DiscoveryModule    ] [packer-virtualbox-iso-1528442324] using discovery type [zen] and seed hosts providers [settings]
[2019-04-23T00:00:02,548][INFO ][o.e.n.Node               ] [packer-virtualbox-iso-1528442324] initialized
[2019-04-23T00:00:02,548][INFO ][o.e.n.Node               ] [packer-virtualbox-iso-1528442324] starting ...
[2019-04-23T00:00:02,636][INFO ][o.e.t.TransportService   ] [packer-virtualbox-iso-1528442324] publish_address {127.0.0.1:9300}, bound_addresses {[::1]:9300}, {127.0.0.1:9300}
[2019-04-23T00:00:02,643][WARN ][o.e.b.BootstrapChecks    ] [packer-virtualbox-iso-1528442324] the default discovery settings are unsuitable for production use; at least one of [discovery.seed_hosts, discovery.seed_providers, cluster.initial_master_nodes] must be configured
[2019-04-23T00:00:02,653][INFO ][o.e.c.c.ClusterBootstrapService] [packer-virtualbox-iso-1528442324] no discovery configuration found, will perform best-effort cluster bootstrapping after [3s] unless existing master is discovered
[2019-04-23T00:00:02,795][INFO ][o.e.c.s.MasterService    ] [packer-virtualbox-iso-1528442324] elected-as-master ([1] nodes joined)[{packer-virtualbox-iso-1528442324}{sjbbwqAqSlqxMWo74zo2PQ}{2Qe-mtCQSTKyp9nbqj1i_g}{127.0.0.1}{127.0.0.1:9300} elect leader, _BECOME_MASTER_TASK_, _FINISH_ELECTION_], term: 2, version: 7, reason: master node changed {previous [], current [{packer-virtualbox-iso-1528442324}{sjbbwqAqSlqxMWo74zo2PQ}{2Qe-mtCQSTKyp9nbqj1i_g}{127.0.0.1}{127.0.0.1:9300}]}
[2019-04-23T00:00:02,809][INFO ][o.e.c.s.ClusterApplierService] [packer-virtualbox-iso-1528442324] master node changed {previous [], current [{packer-virtualbox-iso-1528442324}{sjbbwqAqSlqxMWo74zo2PQ}{2Qe-mtCQSTKyp9nbqj1i_g}{127.0.0.1}{127.0.0.1:9300}]}, term: 2, version: 7, reason: Publication{term=2, version=7}
[2019-04-23T00:00:02,835][INFO ][o.e.g.GatewayService     ] [packer-virtualbox-iso-1528442324] recovered [0] indices into cluster_state
[2019-04-23T00:00:02,839][INFO ][o.e.h.AbstractHttpServerTransport] [packer-virtualbox-iso-1528442324] publish_address {127.0.0.1:9200}, bound_addresses {[::1]:9200}, {127.0.0.1:9200}
[2019-04-23T00:00:02,839][INFO ][o.e.n.Node               ] [packer-virtualbox-iso-1528442324] started
[2019-04-23T00:00:02,977][INFO ][o.e.c.m.MetaDataCreateIndexService] [packer-virtualbox-iso-1528442324] [library] creating index, cause [auto(bulk api)], templates [], shards [1]/[1], mappings []
[2019-04-23T00:00:03,229][INFO ][o.e.c.m.MetaDataMappingService] [packer-virtualbox-iso-1528442324] [library/KW8GCkWKS22L_-uo0JoCfw] create_mapping [book]
[2019-04-23T00:00:03,370][INFO ][o.e.c.m.MetaDataDeleteIndexService] [packer-virtualbox-iso-1528442324] [library/KW8GCkWKS22L_-uo0JoCfw] deleting index
[2019-04-23T00:00:04,186][INFO ][o.e.c.m.MetaDataCreateIndexService] [packer-virtualbox-iso-1528442324] [library] creating index, cause [auto(bulk api)], templates [], shards [1]/[1], mappings []
[2019-04-23T00:00:04,234][INFO ][o.e.c.m.MetaDataMappingService] [packer-virtualbox-iso-1528442324] [library/iQ_gggZtTm-K1SDFnelhVg] create_mapping [book]
[2019-04-23T00:00:04,278][INFO ][o.e.c.m.MetaDataDeleteIndexService] [packer-virtualbox-iso-1528442324] [library/iQ_gggZtTm-K1SDFnelhVg] deleting index
[2019-04-23T00:00:04,296][INFO ][o.e.n.Node               ] [packer-virtualbox-iso-1528442324] stopping ...
[2019-04-23T00:00:04,313][INFO ][o.e.n.Node               ] [packer-virtualbox-iso-1528442324] stopped
[2019-04-23T00:00:04,313][INFO ][o.e.n.Node               ] [packer-virtualbox-iso-1528442324] closing ...
[2019-04-23T00:00:04,321][INFO ][o.e.n.Node               ] [packer-virtualbox-iso-1528442324] closed
"
```

I assume it's trying to find `/usr/lib/jvm/java-8-openjdk-amd64/jre` in the log line that looks similar to this:

```
[2019-04-23T13:51:22,273][INFO ][o.e.n.Node               ] [node-0] JVM home [/usr/lib/jvm/java-8-openjdk-amd64/jre]
```

That log line isn't present at all in the failing log file, presumably because it was logged before midnight and hence when the test checks the contents of the latest log file shortly after midnight it's not in there but in the previous day's log file.

To be robust against log file rolling, tests that check for particular log messages could also check the previous day's logs.

### PR DESCRIPTION
**Pull request by dliappis**
Make JVM presence related packaging tests more resilient, by
also checking for entries in rolled over log files.

Closes #41436

### CODE REVIEW

**Review by alpar-t — COMMENTED** [review]
The changes look good and fix the problem at hand, but we have I think a number of other tests that look at the logs. Could it be that these have the same problem ?  e.x. see search results for  `getServerLog()`

**Comment by dliappis** [comment]
> Could it be that these have the same problem ? e.x. see search results for `getServerLog()`

@atorok Thanks for the heads-up! I wanted to limit this PR for Packaging Tests only, as there is a different framework/library for packaging tests and I think `getServerLog()` is outside of `qa/vagrant/` and address non vagrant test access to rolling logs via a separate issue/PR.

However, I did spot two more occurrences e.g. in [ArchiveTestCase](https://github.com/elastic/elasticsearch/blob/f16ec493d2515e36c806d22e806f5dfb18122f3b/qa/vagrant/src/main/java/org/elasticsearch/packaging/test/ArchiveTestCase.java#L172) and WindowsServiceTestCase so I will take a look at those too. 

Does this make sense?

File: qa/vagrant/src/main/java/org/elasticsearch/packaging/util/FileUtils.java [hunk 1]
```diff
@@ -124,6 +130,41 @@ public static String slurp(Path file) {
...
+            if (e.toString().contains("Not in GZIP format")) {
+                return slurp(file);
+            }
+            throw new RuntimeException(e);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static String slurpAllLogs(Path logPath, String activeLogFile, String rotatedLogFilesGlob) {
```
Comment (rjernst) [inline qa/vagrant/src/main/java/org/elasticsearch/packaging/util/FileUtils.java]: We should have at least a minimal javadoc (just a method description) for any public methods added.

File: qa/vagrant/src/main/java/org/elasticsearch/packaging/util/FileUtils.java [hunk 2]
```diff
@@ -124,6 +130,41 @@ public static String slurp(Path file) {
...
+        StringJoiner logFileJoiner = new StringJoiner("\n");
+        try {
+            logFileJoiner.add(new String(Files.readAllBytes(logPath.resolve(activeLogFile)), StandardCharsets.UTF_8));
+
+            for (Path rotatedLogFile : FileUtils.lsGlob(logPath, rotatedLogFilesGlob)) {
```
Comment (rjernst) [inline qa/vagrant/src/main/java/org/elasticsearch/packaging/util/FileUtils.java]: We should not (probably in the javadoc) that the order of any rolled over logs is not guaranteed. In practice for the packaging tests I don't see how we would ever get more than one, but it's probably still worth noting.

**Review by rjernst — APPROVED** [review]
Looks good, just a couple minor asks. Thanks for picking this up!

File: qa/vagrant/src/main/java/org/elasticsearch/packaging/util/FileUtils.java (hunk 1 above)
Comment (dliappis) [reply qa/vagrant/src/main/java/org/elasticsearch/packaging/util/FileUtils.java]: Addressed in https://github.com/elastic/elasticsearch/pull/44423/commits/69b1678433cf040575f554a23f02f4e87b2f68b3

File: qa/vagrant/src/main/java/org/elasticsearch/packaging/util/FileUtils.java (hunk 2 above)
Comment (dliappis) [reply qa/vagrant/src/main/java/org/elasticsearch/packaging/util/FileUtils.java]: Addressed in https://github.com/elastic/elasticsearch/pull/44423/commits/69b1678433cf040575f554a23f02f4e87b2f68b3

**Pull request merged**